docs/standard/threading/overview-of-synchronization-primitives.md

```diff
@@ -1,174 +1,167 @@
-  
--   <xref:System.Threading.CountdownEvent?displayProperty=nameWithType> represents an event that becomes signaled when its count is zero.  
-  
--   <xref:System.Threading.Barrier?displayProperty=nameWithType> enables multiple threads to synchronize with one another without requiring control by a master thread. A barrier prevents each thread from continuing until all threads have reached a specified point.  
-  
- [Back to top](#top)  
-  
-<a name="spinwait"></a>   
-## SpinWait  
- Starting with the [!INCLUDE[net_v40_short](../../../includes/net-v40-short-md.md)], you can use the <xref:System.Threading.SpinWait?displayProperty=nameWithType> structure when a thread has to wait for an event to be signaled or a condition to be met, but when the actual wait time is expected to be less than the waiting time required by using a wait handle or by otherwise blocking the current thread. By using <xref:System.Threading.SpinWait>, you can specify a short period of time to spin while waiting, and then yield (for example, by waiting or sleeping) only if the condition was not met in the specified time.  
-  
- [Back to top](#top)  
-  
-<a name="interlocked_operations"></a>   
-## Interlocked Operations  
- Interlocked operations are simple atomic operations performed on a memory location by static methods of the <xref:System.Threading.Interlocked> class. Those atomic operations include addition, increment and decrement, exchange, conditional exchange depending on a comparison, and read operations for 64-bit values on 32-bit platforms.  
-  
+# Overview of synchronization primitives
+
+.NET provides a range of types that you can use to synchronize access to a shared resource or coordinate thread interaction.
+
+## WaitHandle class and lightweight synchronization types
+
+Multiple .NET synchronization primitives derive from the <xref:System.Threading.WaitHandle?displayProperty=nameWithType> class, which encapsulates a native operating system synchronization handle. Those classes include:
+
+- <xref:System.Threading.Mutex?displayProperty=nameWithType>
+- <xref:System.Threading.Semaphore?displayProperty=nameWithType>
+- <xref:System.Threading.EventWaitHandle?displayProperty=nameWithType>
+- <xref:System.Threading.AutoResetEvent?displayProperty=nameWithType>
+- <xref:System.Threading.ManualResetEvent?displayProperty=nameWithType>
+
+As <xref:System.Threading.WaitHandle> derives from <xref:System.MarshalByRefObject?displayProperty=nameWithType>, those types can be used to synchronize the activities of threads across application domain boundaries. Moreover, <xref:System.Threading.Mutex>, <xref:System.Threading.Semaphore>, and <xref:System.Threading.EventWaitHandle> can represent named system synchronization handles, which are visible throughout the operating system and can be used for the inter-process synchronization.
```

Comment:
This is .NET Framework specific, .NET Core does not support appdomains. Not sure what's the best way to address such differences in docs.

Reply:
Also, .NET Core on Unix like systems has limited support for named synchronization objects. I think it only support `Mutex`, I need to check.